Content.Server/Mobs/Mind.cs

```diff
@@ -156,9 +156,14 @@ public void RemoveRole(Role role)
         /// <summary>
         /// Adds an objective to this mind.
         /// </summary>
-        public bool TryAddObjective(ObjectivePrototype objective)
+        public bool TryAddObjective(ObjectivePrototype objectivePrototype)
         {
-            if (!objective.CanBeAssigned(this))
+            if (!objectivePrototype.CanBeAssigned(this))
+                return false;
+            if (!objectivePrototype.IsCompatible(_objectives.Select(o => o.Prototype).ToList()))
+                return false;
+            var objective = objectivePrototype.GetObjective(this);
+            if (_objectives.Any(o => objective.Equals(o)))
```

Comment:
Not sure about this, but you could probably use `Contains`? It should use `Equals`